tests/e2e/sample-applications/mongo-persistent/mongo-persistent-block.yaml

```diff
@@ -0,0 +1,267 @@
+apiVersion: v1
+kind: List
+items:
+  - kind: Namespace
+    apiVersion: v1
+    metadata:
+      name: mongo-persistent
+      labels:
+        app: todolist-mongo-go
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: mongo-persistent-sa
+      namespace: mongo-persistent
+      labels:
+        component: mongo-persistent
+  - kind: SecurityContextConstraints
+    apiVersion: security.openshift.io/v1
+    metadata:
+      name: mongo-persistent-scc
+    allowPrivilegeEscalation: true
+    allowPrivilegedContainer: true
+    runAsUser:
+      type: RunAsAny
+    seLinuxContext:
+      type: RunAsAny
+    fsGroup:
+      type: RunAsAny
+    supplementalGroups:
+      type: RunAsAny
+    volumes:
+    - '*'
+    users:
+    - system:admin
+    - system:serviceaccount:mongo-persistent:mongo-persistent-sa
+  - kind: ImageStream
+    apiVersion: image.openshift.io/v1
+    metadata:
+      name: todolist-mongo-go
+      namespace: mongo-persistent
+      labels:
+        app: todolist
+    spec: {}
+    status:
+      dockerImageRepository: ''
+  - apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      annotations:
+        template.alpha.openshift.io/wait-for-ready: 'true'
+      name: mongo
+      namespace: mongo-persistent
+      labels:
+        e2e-app: "true"
+    spec:
+      selector:
+        matchLabels:
+          app: mongo
+      strategy:
+        type: Recreate
+      template:
+        metadata:
+          labels:
+            e2e-app: "true"
+            app: mongo
+        spec:
+          serviceAccountName: mongo-persistent-sa
+          securityContext:
+            runAsUser: 0  
+          # Used to format the block device also touch one file which is used to determine when the
+          # block device was formatted. We use the same image as 
+          initContainers:
```

Comment:
Still did not understand the goal of initContainer, can you explain to me?

Reply:
Please check first comment:
https://github.com/openshift/oadp-operator/pull/1193#issue-1957079616

Also please check the new comment inside the code. Is is still unclear?

Reply:
it creates the filesystem (mkfs.ext4 $DEVICE)  + a file w/ timestamp
q what happens after restore
It will not run the init again, correct?

Reply:
@amastbau After restore it does run init container, but there is logic to only format this once on a first run:
```shell
                  if dumpe2fs -h $DEVICE 2>/dev/null; then
                    echo "Filesystem already exists on $DEVICE"
                  else
                    echo "Formatting $DEVICE"
                    mkfs.ext4 $DEVICE
                  fi
```